RocketChat/Rocket.Chat
# fix: Notifications for thread messages are sent even after thread has been read

## Proposed changes (including videos or screenshots)


 - Clear notification queue after they thread messages have been read by a user;



## Issue(s)


## Steps to test or reproduce

Start a thread with two or more users, and keep it open for both of them while no new message is sent in the main room.
Send a thread message with user A, then read it with user B (by keeping the thread open or opening the thread after the message is sent).
**Current behavior:** a push notification is sent after some minutes;
**Expected behavior:** no push notifications should be sent since the messages have been read (the queue can be checked in the `rocketchat_notification_queue` db collection). Notifications should be cleared when using `getThreadMessages` or `readThreads` methods.

## Further comments

SUP-386
